merlin-rxjava/src/test/java/com.novoda.merlin.rxjava/MerlinObservableShould.java

```diff
@@ -0,0 +1,103 @@
+package com.novoda.merlin.rxjava;
+
+import android.content.Context;
+
+import com.novoda.merlin.Merlin;
+import com.novoda.merlin.NetworkStatus;
+import com.novoda.merlin.registerable.bind.Bindable;
+import com.novoda.merlin.registerable.connection.Connectable;
+import com.novoda.merlin.registerable.disconnection.Disconnectable;
+
+import java.util.List;
+
+import org.junit.Test;
+
+import org.mockito.ArgumentCaptor;
+
+import rx.Subscription;
+import rx.observers.TestSubscriber;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class MerlinObservableShould {
+
+    @Test
+    public void unbindWhenUnsubscribed() {
+        Merlin merlin = mock(Merlin.class);
+        Subscription subscription = MerlinObservable.from(merlin)
+                                                    .subscribe();
+        subscription.unsubscribe();
+
+        verify(merlin, times(1)).unbind();
+    }
+
+    @Test
+    public void receiveOneAvailableNetworkStatusStateOnConnect() {
+        Merlin merlin = mock(Merlin.class);
+        ArgumentCaptor<Connectable> argumentCaptor = ArgumentCaptor.forClass(Connectable.class);
+        TestSubscriber<NetworkStatus.State> testSubscriber = new TestSubscriber<>();
+        MerlinObservable.from(merlin)
+                        .subscribe(testSubscriber);
+        verify(merlin).registerConnectable(argumentCaptor.capture());
+        argumentCaptor.getValue()
+                      .onConnect();
+        List<NetworkStatus.State> stateList = testSubscriber.getOnNextEvents();
+        assertTrue(stateList.size() == 1);
+        assertTrue(stateList.get(0) == NetworkStatus.State.AVAILABLE);
+    }
+
+    @Test
+    public void receiveOneUnavailableNetworkStatusStateOnDisconnect() {
+        Merlin merlin = mock(Merlin.class);
+        ArgumentCaptor<Disconnectable> argumentCaptor = ArgumentCaptor.forClass(Disconnectable.class);
+        TestSubscriber<NetworkStatus.State> testSubscriber = new TestSubscriber<>();
+        MerlinObservable.from(merlin)
+                        .subscribe(testSubscriber);
+        verify(merlin).registerDisconnectable(argumentCaptor.capture());
+        argumentCaptor.getValue()
+                      .onDisconnect();
+        List<NetworkStatus.State> stateList = testSubscriber.getOnNextEvents();
```

Comment:
You can pull these out into a utility method that checks for a single emitted `NetworkStatus`

```
    private void thenEmitsOneNetworkStatusStateOf(NetworkStatus.State state, List<NetworkStatus.State> states) {
        assertTrue(states.size() == 1);
        assertTrue(states.get(0) == state);
    }
```